core/src/main/java/io/questdb/griffin/engine/table/DataFrameRecordCursorFactory.java

```diff
@@ -107,8 +106,16 @@ public PageFrameCursor getPageFrameCursor(SqlExecutionContext executionContext,
         return null;
     }
 
-    public boolean hasDescendingOrder() {
-        return dataFrameCursorFactory.getOrder() == DataFrameCursorFactory.ORDER_DESC;
+    @Override
+    public int getScanDirection() {
+        switch (dataFrameCursorFactory.getOrder()) {
+            case ORDER_ASC:
+                return SCAN_DIRECTION_FORWARD;
+            case ORDER_DESC:
+                return SCAN_DIRECTION_BACKWARD;
+            default:
+                return SCAN_DIRECTION_OTHER;
```

Comment:
This is not covered by tests. I don't think it's possible at the moment. Shall we throw an exception here instead?